pkg/http/http.go

```diff
@@ -144,6 +147,10 @@ func performRequest(req *http.Request, verifyTLS bool, params []queryParam) (int
 	requestErr := retry(5, 100*time.Millisecond, func() error {
 		resp, err := client.Do(req)
 		if err != nil {
+			if resp != nil {
+				defer resp.Body.Close()
+			}
```

Comment:
Small nit, but can we just take this outside of the `if err != nil {` in that case the 2nd defer is not needed anymore (and less likely to introduce a bug in the future)

Reply:
I don't believe we can due to this code executing inside a function. From my understanding, once the function returns the deferred functions will execute, meaning the response body will be closed before we have a chance to read it below.

Reply:
Yup you are right 👍